src/nuid.c

```diff
@@ -208,7 +208,7 @@ natsNUID_init(void)
     natsStatus      s;
     unsigned int    seed = (unsigned int) nats_NowInNanoSeconds();
 
-    memset(&globalNUID, 0, sizeof(natsNUID));
+    memset(&(globalNUID.nuid), 0, sizeof(globalNUID.nuid));
```

Comment:
Good catch in that the memset was wrong, but it was the `sizeof()` that was wrong, not the address of the object. Here, I wanted to initialize `globalNUID`, which is a `natsLockedNUID` object, which itself, is a pointer to a mutex and the natsNUID structure. So the proper initialization is:
```
memset(&globalNUID, 0, sizeof(natsLockedNUID));
```

Reply:
I’ll update, but mutex is initialized one line after, so I used memset() in this way.

Reply:
But that was the point of the memset: if the initialization of the mutex fails, calling natsNUID_free() in case of error would then be dangerous because `globalNUID.mu` pointer may not have been initialized to NULL.